orderer/consensus/etcdraft/node.go

```diff
@@ -276,6 +276,16 @@ func (n *node) abdicateLeader(currentLead uint64) {
 	timer := n.clock.NewTimer(time.Duration(n.config.ElectionTick) * n.tickInterval)
 	defer timer.Stop() // prevent timer leak
 
+	// Revalidate the status before waiting on notifyc channel as Ready go routine
+	// well passed the channel notification before establishing the channel
+	if status.ID != currentLead {
```

Comment:
what if we are a follower and we called `abdicateLeader()` but in the meanwhile there were two leadership changes that resulted in changing from node `i` to `j` and then back to `i` ? Then we won't notice the leadership change. 

Why not instead of looking at who is the leader, we will look at the `Term` from the `status` ? 
After all, each term has its own leader... 

@tock-ibm  what do you think?